lib/linux_admin/network_interface/network_interface_rh.rb

```diff
@@ -0,0 +1,138 @@
+require 'ipaddr'
+require 'pathname'
+
+module LinuxAdmin
+  class NetworkInterfaceRH < NetworkInterface
+    IFACE_DIR = "/etc/sysconfig/network-scripts"
+
+    # @return [Hash<String, String>] Key value mappings in the interface file
+    attr_reader :interface_conf
+
+    # @param interface [String] Name of the network interface to manage
+    def initialize(interface)
+      super
+      @iface_file = Pathname.new(IFACE_DIR).join("ifcfg-#{@interface}")
+      parse_conf
+    end
+
+    # Parses the interface configuration file into the @interface_conf hash
+    def parse_conf
+      @interface_conf = {}
+
+      File.read(@iface_file).each_line do |line|
+        next if line =~ /^\s*#/
+
+        key, value = line.split('=').collect(&:strip)
+        @interface_conf[key] = value
+      end
+      @interface_conf["NM_CONTROLLED"] = "no"
+    end
+
+    # Set the IPv4 address for this interface
+    #
+    # @param address [String]
+    # @raise ArgumentError if the address is not formatted properly
+    def address=(address)
+      validate_ip(address)
+      @interface_conf["BOOTPROTO"] = "static"
+      @interface_conf["IPADDR"]    = address
+    end
+
+    # Set the IPv4 gateway address for this interface
+    #
+    # @param address [String]
+    # @raise ArgumentError if the address is not formatted properly
+    def gateway=(address)
+      validate_ip(address)
+      @interface_conf["GATEWAY"] = address
+    end
+
+    # Set the IPv4 sub-net mask for this interface
+    #
+    # @param mask [String]
+    # @raise ArgumentError if the mask is not formatted properly
+    def netmask=(mask)
+      validate_ip(mask)
+      @interface_conf["NETMASK"] = mask
+    end
+
+    # Sets one or both DNS servers for this network interface
+    #
+    # @param servers Array<String> The DNS servers
+    def dns=(servers)
+      @interface_conf["DNS1"] = servers[0]
+      @interface_conf["DNS2"] = servers[1] if servers[1]
+    end
+
+    # Sets the search domain list for this network interface
+    #
+    # @param domains [Array<String>] the list of search domains
+    def search_order=(domains)
+      @interface_conf["DOMAIN"] = "\"#{domains.join(' ')}\""
+    end
+
+    # Set up the interface to use DHCP
+    # Removes any previously set static networking information
+    def enable_dhcp
+      @interface_conf["BOOTPROTO"] = "dhcp"
+      @interface_conf.delete("IPADDR")
+      @interface_conf.delete("NETMASK")
+      @interface_conf.delete("GATEWAY")
+      @interface_conf.delete("PREFIX")
+      @interface_conf.delete("DNS1")
+      @interface_conf.delete("DNS2")
+      @interface_conf.delete("DOMAIN")
+    end
+
+    # Applies the given static network configuration to the interface
+    #
+    # @param ip [String] IPv4 address
+    # @param mask [String] subnet mask
+    # @param gw [String] gateway address
+    # @param dns1 [Array<String>] list of dns servers
+    # @param search [Array<String>] list of search domains
```

Comment:
Looks like yardoc takes care of it, this is what is shows:
```
 search (Array<String>) (defaults to: nil) — list of search domains
```

Is that good enough or do you also want me to put a note about it being optional in there?